ecl/provider_connectivity/v2/tenant_connection_requests/requests.go

```diff
@@ -0,0 +1,130 @@
+	TenantIDOther             string `q:"tenant_id_other"`
+	NetworkID                 string `q:"network_id"`
+	ApprovalRequestID         string `q:"approval_request_id"`
+}
+
+// ToTenantConnectionRequestListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToTenantConnectionRequestListQuery() (string, error) {
+	q, err := eclcloud.BuildQueryString(opts)
+	if err != nil {
+		return "", err
+	}
+	return q.String(), nil
+}
+
+// List retrieves a list of Tenant Connection Requests.
+func List(client *eclcloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(client)
+	if opts != nil {
+		query, err := opts.ToTenantConnectionRequestListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return TenantConnectionRequestPage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// Get retrieves details of an Tenant Connection Request.
+func Get(client *eclcloud.ServiceClient, id string) (r GetResult) {
+	_, r.Err = client.Get(getURL(client, id), &r.Body, nil)
+	return
+}
+
+// CreateOptsBuilder allows extensions to add additional parameters to
+// the Create request.
+type CreateOptsBuilder interface {
+	ToTenantConnectionRequestCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts provides options used to create a Tenant Connection Request.
+type CreateOpts struct {
+	TenantIDOther string            `json:"tenant_id_other"`
+	NetworkID     string            `json:"network_id"`
+	Name          string            `json:"name,omitempty"`
+	Description   string            `json:"description,omitempty"`
+	Tags          map[string]string `json:"tags,omitempty"`
+}
+
+// ToTenantConnectionRequestCreateMap formats a CreateOpts into a create request.
+func (opts CreateOpts) ToTenantConnectionRequestCreateMap() (map[string]interface{}, error) {
+	return eclcloud.BuildRequestBody(opts, "tenant_connection_request")
+}
+
+// Create creates a new Tenant Connection Request.
+func Create(client *eclcloud.ServiceClient, opts CreateOptsBuilder) (r CreateResult) {
+	b, err := opts.ToTenantConnectionRequestCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = client.Post(createURL(client), &b, &r.Body, &eclcloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// Delete deletes a Tenant Connection Request.
+func Delete(client *eclcloud.ServiceClient, id string) (r DeleteResult) {
+	_, r.Err = client.Delete(deleteURL(client, id), &eclcloud.RequestOpts{
+		OkCodes: []int{204},
+	})
+	return
+}
+
+// UpdateOptsBuilder allows extensions to add additional parameters to
+// the Update request.
+type UpdateOptsBuilder interface {
+	ToTenantConnectionRequestUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts represents parameters to update a Tenant Connection Request.
+type UpdateOpts struct {
+	Name             string            `json:"name,omitempty"`
```

Comment:
If users want to update name filed to blank string, it will be omitted with current code. That's why you need to make it pointer.
```suggestion
	Name             *string            `json:"name,omitempty"`
```

Reply:
There was no such consideration. Fixed.

Reply:
Can you covert test case that the filed is nil and blank string?

Reply:
Added a test case when the field is blank or nil.